Orange/widgets/data/owcreateinstance.py

```diff
@@ -646,12 +647,34 @@ def _create_data_from_values(self) -> Table:
                 data[:, var_name] = value
         return data
 
-    def _append_to_data(self, data: Table) -> Table:
+    def _append_to_data(self, instance: Table) -> Table:
         assert self.data
-        assert len(data) == 1
+        assert len(instance) == 1
+        source_label = "__source_widget"
 
-        var = DiscreteVariable("Source ID", values=(self.data.name, data.name))
-        data = Table.concatenate([self.data, data], axis=0)
+        data = Table.concatenate([self.data, instance], axis=0)
+        for var in self.data.domain.variables + self.data.domain.metas:
```

Comment:
What about replacing all those `if`-s with `get_column_view`? I think (and also tried) that this should work:

```python
        for var in chain(self.data.domain.variables, self.data.domain.metas):
            if var.attributes.get(source_label) == OWCreateInstance:
                with data.unlocked():
                    data.get_column_view(var)[0][-1] = 1
                return data
```

Reply:
done